public/app/features/browse-dashboards/components/BulkActions/utils.ts

```diff
@@ -16,12 +16,64 @@ export interface BulkActionProvisionResourceProps {
   onDismiss?: () => void;
 }
 
-export function getTargetFolderPathInRepo({ targetFolder }: { targetFolder?: Folder }): string | undefined {
+/**
+ * @example
+ * // Whole instance provisioned (root)
+ * getTargetFolderPathInRepo({ targetFolderUID: '' }) // returns "/"
+ *
+ * // Repository root folder
+ * getTargetFolderPathInRepo(...) // returns "/"
+ *
+ * // Nested folder
+ * getTargetFolderPathInRepo(...) // returns "path/to/folder/"
+ */
+export function getTargetFolderPathInRepo({
+  targetFolderUID,
+  targetFolder,
+  repoName,
+  hidePrependSlash = false,
+}: {
+  targetFolderUID?: string;
+  targetFolder?: Folder;
+  repoName?: string;
+  hidePrependSlash?: boolean;
+}): string | undefined {
+  const ROOT_PATH = '/';
+  const EMPTY_PATH = '';
+
+  // Case 1: Whole instance is provisioned and no folder uid passed in (empty UID indicates root)
+  if (targetFolderUID === '') {
+    return hidePrependSlash ? EMPTY_PATH : ROOT_PATH;
+  }
+
+  // Case 2: Target folder is the repository root folder
+  if (isRepositoryRootFolder(targetFolder, repoName)) {
+    return hidePrependSlash ? EMPTY_PATH : ROOT_PATH;
+  }
+
+  // Case 3: Regular folder with source path annotation
+  return getNestedFolderPath(targetFolder);
+}
+
+/**
+ * Checks if the target folder is the repository root folder
+ */
+function isRepositoryRootFolder(targetFolder?: Folder, repoName?: string): boolean {
```

Comment:
```suggestion
function isRepositoryRootFolder(targetFolder?: Folder, repoName?: string) {
```
By relying on inferred type returns (for simple types) we can make the code a bit easier to maintain.